src/Adapter/Module/ModuleDataProvider.php

```diff
@@ -370,4 +368,18 @@ private function getDeviceStatus($name)
 
         return false;
     }
+
+    /**
+     * Checks if the module is active on at least one shop of the context.
+     */
+    private function isModuleActive(int $id, array $id_shops): bool
+    {
+        $result = Db::getInstance()->getRow('SELECT m.`active`, ms.`id_module` as `shop_active`, ms.`enable_device` as `enable_device`
+            FROM `' . _DB_PREFIX_ . 'module` m
+            LEFT JOIN `' . _DB_PREFIX_ . 'module_shop` ms ON m.`id_module` = ms.`id_module`
+            WHERE m.`id_module` = ' . $id . '
+            AND ms.`id_shop` IN (' . implode(',', $id_shops) . ')');
+
+        return !empty($result);
+    }
```

Comment:
Should we check if m.active equals 1 ?

Reply:
this is a backport of this PR: https://github.com/PrestaShop/PrestaShop/pull/31379/commits

Reply:
see line 163, we replace the `active` value of the module with the result of this function. Because a module is considered active if it is activated for at least one shop.